data_augmentation.py

```diff
@@ -0,0 +1,238 @@
+    ## original image and keypoints 
+    # aug_pipeline_det = aug_pipeline.to_deterministic()
+
+    return aug_pipeline
+
+# %% init augmentation pipeline
+# NOTE: DO NOT APPLY AFFINE SCALE AFTER CROP
+# Keypoints that should have disappeared will be part of the image again:
+# https://github.com/aleju/imgaug/issues/187
+def heavy_augmentation():
+    # Perform data augmentation randomly
+            # - Rotation (+/- 30 deg)
+            # - Scaling (.75 to 1.25)
+            # - Horizontal flip (left to right) with probability 50%
+            # - Gaussian noise 
+            # - Random brightness
+            # - Random gamma (contrast)
+            # - Random dropout (fine dropout. Coarse may drop out the area where our keypoint is)
+
+    # define an augmentation pipeline
+    aug_pipeline = iaa.Sequential([
+        iaa.Sometimes(0.3, iaa.GaussianBlur((0, 2.0))), # apply Gaussian blur with a sigma between 0 and 2 to 30% of the images # used to be [0,3] on 50% images
+        iaa.Sometimes(0.5, iaa.Fliplr(1.0)), # horizontally flip 50% of the time
+        # apply from 0 to 3 of the augmentations from the list
+        iaa.SomeOf((0, 3),[
+            iaa.Dropout((0, 0.1), per_channel=0.5), # randomly remove up to 5% of the pixels # used to be 10%
+            iaa.Sharpen(alpha=(0, 1.0), lightness=(0.75, 1.5)), # sharpen images
+            iaa.AddToHueAndSaturation((-30, 30)),  # change their color #(-60, 60)
+            iaa.OneOf([
+                iaa.AddToBrightness((-30,30)),
+                # Strengthen or weaken the contrast in each image.
+                iaa.LinearContrast((0.75, 1.5)),
+            ]),
+            # Add gaussian noise.
+            # For 50% of all images, we sample the noise once per pixel.
+            # For the other 50% of all images, we sample the noise per pixel AND
+            # channel. This can change the color (not only brightness) of the
+            # pixels.
+            iaa.AdditiveGaussianNoise(loc=0, scale=(0.0, 0.05*255), per_channel=0.5),
+        ]),
+        # Only apply one of the following because otherwise there is a risk that keypoints will 
+        # be pointing to a non-existent part of the image
+        iaa.SomeOf((0,1),[
+            iaa.CropAndPad(percent=(-0.25, 0.25), keep_size=True, sample_independently=False), # crop and pad 50% of the images
+            iaa.Rotate((-30,30)), # rotate 50% of the images between [-30, 30] degrees
+        ])
+    ],
+    random_order=True # apply the augmentations in random order
+    )
+
+    ## We may need this line so our pipeline will apply the same operations to keypoints as well as images
+    ## Disabled because we will just pass both the keypoints and the image to the seq pipeline
+    ## which eliminates the possibility that different augmentations will be applied on the 
+    ## original image and keypoints 
+    # aug_pipeline_det = aug_pipeline.to_deterministic()
+
+    return aug_pipeline
```

Comment:
I'm working to refactor this code into something that's easier to read. Stay tuned.